c++/src/capnp/rpc.c++

```diff
@@ -3030,6 +3079,16 @@ private:
 
     // No more using `call` after this point, as it now belongs to the context.
 
+    if (call.getIsRealtime()) {
+      auto cancelPaf = kj::newPromiseAndFulfiller<void>();
+      auto promiseAndPipeline = startCall(
```

Comment:
Problem: `hints.isRealtime` may be true here, but `startCall()` does not know whether it's making a streaming call. If `capability` itself turns out to be an outgoing RPC client, i.e. the call is going to be proxied over another connection, then ultimately `RpcRequest::send()` will be invoked to send the call. The outgoing call will have `isRealtime = true` due to the call hint, but because `send()` is used to send it, a regular question table entry will be created, with a regular low-numbered question ID. Since it's a realtime call, the recipient may never send a `Return` back, so the question table entry will leak.

To fix this, I think `RpcRequest::send()` needs to check if the `isRealtime` bit is set and, if so, delegate to `sendStreaming()` (or maybe directly to `sendRealtimeInternal()` depending on what makes the code simpler).

Please be sure to add a test for this case, that is, arrange a proxy so that a capability crosses two connections, and then try doing realtime calls on it. You'll need some way to detect if question IDs are being leaked... maybe you can add a `RpcSystem::countQuestionsForTest()` that returns a count of all outstanding questions on all connections?